API/API/Controllers/ProductController.cs

```diff
@@ -1,109 +1,96 @@
+    /// Get product data
     /// </summary>
-    [Route("/Products")]
-    [ApiController]
-    public class ProductController:ControllerBase
+    /// <param name="GetProduct"> product Id </param>
+    /// <returns> product Id  </returns>
+    [HttpGet("{GetProduct}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<ActionResult<ProductModel>> GetProduct(int GetProduct)
     {
-        private readonly ProductContext _context;
-
-        public ProductController(ProductContext context)
+        try
         {
-            _context = context;
+            ProductModel product = await _productService.GetProduct(GetProduct);
+            return Ok(product);
         }
-
-        /// <summary>
-        /// Get product data
-        /// </summary>
-        [HttpGet]
-        public async Task<ActionResult<ProductItem>> GetProductItem(int id)
+        catch (ExceptionClass ex)
         {
-            var productItem = await _context.ProductItems.FindAsync(id);
-
-            if (productItem == null)
-            {
-                return NotFound();
-            }
-            return productItem;
+            return BadRequest(ex.Message);
         }
+    }
 
-        /// <summary>
-        /// Post product data
-        /// </summary>
-        [HttpPost]
-        public async Task<ActionResult<ProductItem>> PostProductItem(ProductItem productItem)
+    /// <summary>
+    /// Post product data
+    /// </summary>
+    [HttpPost]
+    public async Task<ActionResult<ProductModel>> PostProduct(ProductModel PostProduct)
+    {
+        try
         {
-
-            _context.ProductItems.Add(productItem);
-            await _context.SaveChangesAsync();
-
-            return CreatedAtAction(nameof(GetProductItem), new { id = productItem.Id }, productItem);
+            await _productService.CreateProduct(PostProduct);
+            return Ok(PostProduct);
         }
-
-        /// <summary>
-        /// Put product data
-        /// </summary>
-        [HttpPut]
-        public async Task<IActionResult> PutProductItem(int id,ProductItem productItem)
+        catch (ExceptionClass ex)
         {
-            if (id != productItem.Id)
-            {
-                return BadRequest();    
-            }
-
-            _context.Entry(productItem).State= EntityState.Modified;
+            return BadRequest(ex.Message);
+        }
+    }
 
-            try
-            {
-                await _context.SaveChangesAsync();
-            }
-            catch(DbUpdateConcurrencyException)
-            {
-                if (!ProductExists(id))
-                {
-                    return NotFound();
-                }
-                else
-                {
-                    throw;
-                }
-            }
-            //return NoContent();
-            return CreatedAtAction(nameof(GetProductItem), new { id = productItem.Id }, productItem);
+    /// <summary>
+    /// Put product data
+    /// </summary>
+    [HttpPut]
```

Comment:
Is a good pracitce to request de id in the path and the data in the body of the request: products/{idProduct}

Reply:
if I required all the model for HttpPut, I have to request de Id or all the paremeters?

Reply:
Done